pyogrio/tests/test_geopandas_io.py

```diff
@@ -186,6 +186,124 @@ def test_read_fids_force_2d(test_fgdb_vsi):
         assert not df.iloc[0].geometry.has_z
 
 
+@pytest.mark.filterwarnings("ignore: Layer")
+def test_read_sql(naturalearth_lowres):
+    # The geometry column cannot be specified when using the 
+    # default OGRSQL dialect but is returned nonetheless, so 4 columns.
+    sql = "SELECT iso_a3 AS iso_a3_renamed, name, pop_est FROM naturalearth_lowres"
+    df = read_dataframe(naturalearth_lowres, sql=sql, sql_dialect="OGRSQL")
+    assert len(df.columns) == 4
+    assert len(df) == 177
+
+    # Columns and/or where filters are applied after the sql query, so alias 
+    # needs to be specified instead of original column name.
+    df = read_dataframe(
+        naturalearth_lowres, 
+        sql=sql, 
+        sql_dialect="OGRSQL", 
+        columns=["iso_a3_renamed", "name"],
+        where="iso_a3_renamed IN ('CAN', 'USA', 'MEX')",
+    )
+    assert len(df.columns) == 3
+    assert len(df) == 3
+    assert len(set(df.iso_a3_renamed.unique()).difference(["CAN", "MEX", "USA"])) == 0
```

Comment:
```suggestion
    assert df.iso_a3_renamed.tolist == ["CAN", "MEX", "USA"]
```

Suggest similar changes below to sidestep set intersections.

(update order, if needed)